modules/aggregation-layers/src/contour-layer/contour-layer.js

```diff
@@ -218,6 +189,21 @@ export default class ContourLayer extends GridAggregationLayer {
     return strokeWidth;
   }
 
+  _updateSubLayerTriggers(oldProps, props) {
+    if (oldProps && oldProps.contours && props && props.contours) {
+      if (props.contours.some((contour, i) => contour.color !== oldProps.contours[i].color)) {
+        this.state.colorTrigger++;
```

Comment:
Is this really useful? If the user defines contours inline, the color array will for sure change.

To remove complexity, you could just do shallow comparison here and use one state `contourVersion` to track changes.

We don't do deep comparison in any other layers. It still incurs a perf cost even if we skip attribute updates here. It's more important to promote best practice (avoid shallow change in props) by e.g. updating the docs and examples to use constant contour definitions.

Reply:
Addressed in https://github.com/uber/deck.gl/pull/3954
Added shallow compare of `contours` prop, when it changes, contours are regenerated resulting sublayers data change, hence triggers are not needed too.